src/main/java/hudson/plugins/ec2/EC2RetentionStrategy.java

```diff
@@ -139,23 +139,23 @@ private long internalCheck(EC2Computer computer) {
             }
         }
 
-        if (computer.isIdle() && !DISABLED) {
-            final long uptime;
-            final Instant launchedAt;
-            InstanceState state;
-
-            try {
-                state = computer.getState(); // Get State before Uptime because getState will refresh the cached EC2
-                // info
-                uptime = computer.getUptime();
-                launchedAt = computer.getLaunchTime();
-            } catch (SdkException | InterruptedException e) {
-                // We'll just retry next time we test for idleness.
-                LOGGER.fine("Exception while checking host uptime for " + computer.getName()
-                        + ", will retry next check. Exception: " + e);
-                return CHECK_INTERVAL_MINUTES;
-            }
+        final long uptime;
+        final Instant launchedAt;
+        InstanceState state;
+
+        try {
+            state = computer.getState(); // Get State before Uptime because getState will refresh the cached EC2
+            // info
+            uptime = computer.getUptime();
+            launchedAt = computer.getLaunchTime();
+        } catch (SdkException | InterruptedException e) {
+            // We'll just retry next time we test for idleness.
+            LOGGER.fine("Exception while checking host uptime for " + computer.getName()
+                    + ", will retry next check. Exception: " + e);
+            return CHECK_INTERVAL_MINUTES;
+        }
 
+        if (computer.isIdle()) {
```

Comment:
Not sure I see the point of this part of the patch. `state` / `uptime` / `launchedAt` are still used only if `idle`, so why calculate them if it is not?

Reply:
Yeah, I actually pulled them out when I refactored teh code differently. I'll move them back in